### ISSUE TITLE
DeepSeek-OCR: NaN logits (CPU forge + TT) with transformers 5.2; 4.46 finite

### ISSUE THREAD

**Issue by ashokkumarkannan1** [tenstorrent/tt-xla#4703]
### Describe the bug

- TT / compile path: DeepSeek-OCR (DeepseekOCRForCausalLM via tt-forge-models + tt-xla) shows NaNs and/or poor PCC on device runs (whole model  - [log](https://github.com/user-attachments/files/27770370/deepseek_ocr_log.log))
- CPU reference (forge, local ModelLoader): The same forge forward (ModelLoader + forked modeling_* under third_party/tt_forge_models) is not numerically stable across transformers major pins
- transformers==4.46.3 (current HF model supported version): one forward produces all-finite logits
- transformers==5.2.0(current tt-xla env version): the same script path produced all-NaN logits before compatibility fixes; 5.2 also changes LlamaAttention.forward to return three values (attn_output, attn_weights, past_key_value) while local DeepseekV2DecoderLayer assumed two, causing unpack / graph issues until fixed in modeling_deepseekv2.py.
- HF Hub reference (deepseek-ai/DeepSeek-OCR, trust_remote_code): Checkpoint metadata targets ~4.46 era Transformers. 
a) On 5.2.x, hub + test harness needed several fixes: AutoModel (hub auto_map maps AutoModel → DeepseekOCRForCausalLM, not AutoModelForCausalLM), tests.infra / infra duplicate-package circular import (sys.modules alias)
b) CacheLayerMixin / MLACache compatibility in tests/infra/utilities/torch_mla_cache.py, and hub modeling_deepseekocr.py calling .cuda() on masks under CPU-only PyTorch (workaround: temporary Tensor.cuda identity when CUDA is unavailable). After that, HF and forge are finite on CPU, while still diverging in value (low PCC)

### Key Observations:
- Transformers pin is a confirmed major lever on CPU forge reference numerics (finite vs all-NaN logits in controlled script runs between 4.46.3 and 5.2.0).
- HF hub + 5.2 is workable only with explicit harness/modeling fixes; HF vs forge PCC is still poor when both are finite—do not equate “no NaN” with “same reference.”

### Steps to reproduce the issue

```
git checkout akannan/deepseek_nan_check
git submodule update --init --recursive

# A/B Transformers (use separate venvs or reinstall between runs)
pip install 'transformers==4.46.3'
PYTHONPATH=. python scripts/local_deepseek_ocr_forward_nan_check_torch52.py --expect-transformers '4.46'

pip install 'transformers==5.2.0'
PYTHONPATH=. python scripts/local_deepseek_ocr_forward_nan_check_torch52.py --expect-transformers '5.2'

# HF vs forge (CPU-only script; downloads hub + uses local snapshot)
PYTHONPATH=. python scripts/compare_deepseek_ocr_hf_vs_forge_loader.py

# Pytest HF sanity
pytest -q tests/torch/models/deepseek_ocr/test_hf_forward_sanity.py

```

**Comment by ashokkumarkannan1** [comment]
this [fix](https://github.com/tenstorrent/tt-forge-models/pull/695) solves the issue
